src/dmd/globals.d

```diff
@@ -572,6 +574,15 @@ nothrow:
         this.filename = filename;
     }
 
+    //version(ExtraLocation)
+        extern (D) this(const(char)* filename, uint linnum, uint charnum, uint offset) pure
```

Comment:
It's probably better to simply add the parameter by default and discard the value without `version(ExtraLocation`)